### PR TITLE
Documenter

### PR DESCRIPTION
**Pull request by CarloLucibello**
rebase of #366, plus some polishing @hayd 

@sbromberger , after merging this you should follow  

https://juliadocs.github.io/Documenter.jl/stable/man/hosting/

to set up the automatic doc building. I think thant only the Travis.genkeys("MyPackage") part is needed, and maybe some adjustament in the config of the hosting page on readthedocs

### CODE REVIEW

**Comment by sbromberger** [comment]
This seems to be a big mess of commits. What's the purpose of this PR?

**Comment by CarloLucibello** [comment]
to move the docs system to Documenter.jl

**Comment by sbromberger** [comment]
ok, so what's the commit to change modularity?

**Comment by CarloLucibello** [comment]
I glimpsed a cause of type instability in the modularity file, the initialization `Q=0`, so I changed it to `Q=0.`. This caused some trubles in the rebase that I add to fix

**Comment by sbromberger** [comment]
This now violates our first rule for PRs. One change per PR, even if they're both good.

**Comment by CarloLucibello** [comment]
ooook, it removed the extra `.` in modularity, can we merge this now?

**Comment by sbromberger** [comment]
Thanks. I'm not opposed to merging this but I need to understand how to regenerate (and host) the docs before I do, and given my workload this probably won't happen for a couple of weeks.

**Comment by CarloLucibello** [comment]
automatic doc building is taken care in 
```julia
julia -e 'cd(Pkg.dir("LightGraphs")); include(joinpath("docs", "make.jl"))' 
```
in travis.yml. As for hosting it should be really easy since we already have the hosting space set up. I wouldn't be surprised if things worked out just merging this, but I am really no expert in this hosting stuff. Anyway take your time.

File: src/LightGraphs.jl [hunk 1]
```diff
@@ -138,8 +135,8 @@ include("core.jl")
             include("persistence/graphml.jl")
             include("persistence/net.jl")
             include("persistence/jld.jl")
-        include("randgraphs.jl")
-        include("generators.jl")
+        include("generators/staticgraphs.jl")
+            include("generators/randgraphs.jl")
```
Comment (sbromberger) [inline src/LightGraphs.jl]: Why are we doing this in a doc update PR?

Comment (CarloLucibello) [inline src/LightGraphs.jl]: to have the docs, where there is a single page, graph generators,documenting static graphs and randgraphs,  coherent with the folder structure

**Comment by juliohm** [comment]
@CarloLucibello thank you for putting the effort towards Documenter.jl, I was about to ask why LG is not using it.

**Comment by sbromberger** [comment]
I think we're almost ready to do this - I need to understand 1) where the docs will be living once we merge this, and 2) what the new process is to generate documentation.

**Comment by CarloLucibello** [comment]
thanks @juliohm, most credits go to @hayd though

@sbromberger
> I think we're almost ready to do this - I need to understand 1) where the docs will be living once we merge this, and 2) what the new process is to generate documentation.

the docs live in the docs/ folde. They are automatically compiled at each commit (see last line of travis.yml) form the files in docs/src. Compiled doc files will be in docs/build

see also https://juliadocs.github.io/Documenter.jl/stable/man/hosting.html

We should give a try at merging this and see how it goes, it could take a few trials to get the automatic building and hosting right, but I couldn't tell without trying. 

At least this won't break anything, worst case we will have https://github.com/JuliaGraphs/LightGraphs.jl  down for  a few hours.

https://github.com/JuliaGraphs/LightGraphs.jl should be pointed to `docs/build` instead of the old `doc`

**Comment by sbromberger** [comment]
OK, we should definitely do this. @CarloLucibello - are you available at a specific time to hang out on gitter and work through any post-merge issues? We can coordinate in real time to minimize disruption.

**Comment by CarloLucibello** [comment]
what about now?

**Comment by sbromberger** [comment]
works for me. See you on gitter / LightGraphs?

**Pull request merged**